controllers/external/util.go

```diff
@@ -84,10 +84,75 @@ type CreateFromTemplateInput struct {
 	// Annotations is an optional map of annotations to be added to the object.
 	// +optional
 	Annotations map[string]string
+
+	// SSA when set to true will create the object using server side apply.
+	// When SSA is set to ture, SSAPatchOptions cannot be empty.
+	SSA bool
+
+	// SSAPatchOptions is used when using SSA to creat the object.
+	// The value is ignored when SSA is set to false.
+	SSAPatchOptions []client.PatchOption
 }
 
 // CreateFromTemplate uses the client and the reference to create a new object from the template.
 func CreateFromTemplate(ctx context.Context, in *CreateFromTemplateInput) (*corev1.ObjectReference, error) {
```

Comment:
I think it would be cleaner to add a new ApplyFromTemplate.

Preserves the current CreateFromTemplate as is and makes it less awkward to use CreateFromTemplate for updates/applies